### PR TITLE
Add custom CSS settings to satspay server

### PR DESCRIPTION
**Pull request by blackcoffeexbt**
This PR adds the ability for users to specifiy their own custom CSS to be used with for their SatsPay invoice pages.

### CODE REVIEW

File: lnbits/extensions/satspay/migrations.py [hunk 1]
```diff
@@ -26,3 +26,18 @@ async def m001_initial(db):
...
+
+async def m002_add_settings_table(db):
+    """
+    Settings table
+    """
+
+    await db.execute(
+        """
+        CREATE TABLE satspay.settings (
+            "user" TEXT,
```
Comment (dni) [inline lnbits/extensions/satspay/migrations.py:39]: i would go for `user_id` here, it is more descriptive

File: lnbits/extensions/satspay/templates/satspay/index.html [hunk 2]
```diff
@@ -642,6 +705,24 @@ <h6 class="text-subtitle1 q-my-none">
           this.rescanning = false
         }
       },
+      saveSettings: async function (wallet, data) {
+        console.log('wallet', wallet)
```
Comment (dni) [inline lnbits/extensions/satspay/templates/satspay/index.html:709]: logs

**Comment by blackcoffeexbt** [comment]
Updated as per feedback from @dni

File: lnbits/extensions/satspay/crud.py [hunk 3]
```diff
@@ -129,3 +129,44 @@ async def get_charge_config(charge_id: str):
         """SELECT "user" FROM satspay.charges WHERE id = ?""", (charge_id,)
     )
     return await get_config(row.user)
+
+
+################## SETTINGS ###################
+async def save_settings(user_id: str, data: SatsPaySettings):
+    # insert or update
```
Comment (motorina0) [inline lnbits/extensions/satspay/crud.py:136]: - could use `ON CONFLICT` here
Eg:

```py
async def update_user_extension(
    *, user_id: str, extension: str, active: bool, conn: Optional[Connection] = None
) -> None:
    await (conn or db).execute(
        """
        INSERT INTO extensions ("user", extension, active) VALUES (?, ?, ?)
        ON CONFLICT ("user", extension) DO UPDATE SET active = ?
        """,
        (user_id, extension, active, active),
    )
```

**Comment by motorina0** [comment]
Any reason why not to have the custom CSS per `charge` (instead of per user)?
It is possible that a user wants to offer different CSS for different charges?

**Comment by arcbtc** [comment]
And when they create the charge they can select the template? Nice.
This could be a future update perhaps, for now what @blackcoffeexbt has done is a great update.

**Comment by arcbtc** [comment]
Actually. going to add Vlads suggestion, as doing now makes sense

**Comment by arcbtc** [comment]
Fixed a few bugs and updated so now you can have multiple themes
image

**Comment by arcbtc** [comment]
I totally broke trying to resolve conflicts after latest satspay update :(
Once I have resolved, it would be good if you can take a look @motorina0 to make sure I have not stepped on your code anywhere

**Comment by arcbtc** [comment]
All fixed. Could Vlad check to make sure the functionality he added has not been affected, it was a messy conflict resolve for some reason.

**Comment by motorina0** [comment]
@arcbtc I guess this PR can be closed now (changes merged whit https://github.com/lnbits/lnbits/pull/1153)

**Pull request closed without merging**